From: Sunil Mushran <sunil.mushran@oracle.com>
Date: Wed, 4 May 2011 10:28:07 -0700
Subject: [PATCH] ocfs2/dlm: Target node death during resource
 migration leads to thread spin
Patch-mainline: yes
Git-commit: df016c665b10ae80d8db67ec8103b50c5c234e5c

During resource migration, if the target node were to die, the thread doing
the migration spins until the target node is not removed from the domain map.
This patch slows the spin by making the thread wait for the recovery to kick in.

Signed-off-by: Sunil Mushran <sunil.mushran@oracle.com>
Signed-off-by: Joel Becker <jlbec@evilplan.org>
Acked-by: David Sterba <dsterba@suse.cz>
---
 fs/ocfs2/dlm/dlmmaster.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/fs/ocfs2/dlm/dlmmaster.c
+++ b/fs/ocfs2/dlm/dlmmaster.c
@@ -2576,6 +2576,9 @@ fail:
 		res->state &= ~DLM_LOCK_RES_MIGRATING;
 		wake = 1;
 		spin_unlock(&res->spinlock);
+		if (dlm_is_host_down(ret))
+			dlm_wait_for_node_death(dlm, target,
+						DLM_NODE_DEATH_WAIT_MAX);
 		goto leave;
 	}
 
